### ISSUE TITLE
No .html file when creating a module

### ISSUE THREAD

**Issue by corneadoug** [cgross/generator-cg-angular#105]
I find it pretty strange for a 'module' not to have it's own .html file.
You will obviously cut your module into multiple partials, but you still need to define the ui-view or ng-include for your partials. (What better place than the module .html?)

**Comment by cgross** [comment]
I have no idea what sense it would make to have a random html file created when you create a module.  If you want a partial, create one in the module...?   Why would you declare a ui-view in a module?  Wouldn't that go in the main app (likely the index.html)?

**Comment by corneadoug** [comment]
Maybe I wasn't concise enough,

When you create your angular app, you will have your app module in app.js, and your index.html file.

From the angular Module Documentation: 'Most applications have a main method that instantiates and wires together the different parts of the application.'

I think that it should not only apply to .js files, just like index.html is linking other partials of the app module, then having a .html file in a module would link its own partials.

For big applications, especially those needing ui-router, you will start breaking your application modules base on your main 'pages'.

For example, if you visit mywebsite.com/admin, you would see your admin page, which is composed of multiple partials which are all connected together in admin.html.

admin.html
```
<div>
  <div ng-include="admin-partial1/admin-partial1.html"></div>
  <div ng-include="admin-partial2/admin-partial2.html"></div>
  <div ng-include="adming-partial3/admin-partial3.html"></div>
</div>
```

In the current module configuration, we need to create a first partial to do the job of admin.html (admin-partial), then create all the other partials into admin-partial if we want to respect some kind of logical directory structure. Which in the end is not super elegant.

Maybe it is only a personal choice, juste like the fact that I like having a main controller in a module (and I can simply add it inside app.js or my_module.js).

I can of course just create by hand that .html file, I just thought it could make sense to have it directly in the generator.

**Comment by eduardoinnorway** [comment]
I don't agree at all, this will really make a messy application. Now working in 2 big complex projects using cg-angular and before that also working with generator-angular, I have to say the modular approach is the winner. The issue here is what is when to make a module or not, and what is a module. This is how I see things.

Using UI-router I defined a structure of my application, that can be directly translated to the "partials" how cg-angular uses them.

So myApp can be like this:
 state('app') //Has footer and header
 state('app.main') //Has MainCtrl and MainView, then app.admin and so on.
 .... etc.

Then the folder structure will be something like this:

/partials/
   /commons/ // Has header, footer and other commons
  /home/ //Has HomeCtrl, html, test

Now Home can have many Views and Controllers, that are like multiple partials. For this, there is not needed to have them here, because they should belong to a Module if this is correctly designed.

For example, Home has "Top 10 posts table", "Latest News" and "Welcome User box". So the choices you have now is to you can have a huge home HTML  with everything or you can make partials to include or directives instead. I prefer directives.

So here comes the strong part of cg-angular approach, that Post, News, User, etc.. all are modules and everything related to them such as Directives, Services, Views corresponding should go in them and not spread it out all over your application.

Then we would have for example:
/modules/
  /posts/ //postService, post directive(10 top posts), partials (10 top posts html) etc..
  /news/
  / user/ ... etc... 

So reading up on what you are referring to, Module should NOT have its own .html file. Embrace cg-angular, it really rocks.
